### PR TITLE
Options refactor

### PR DESCRIPTION
**Pull request by JensWehner**
Implementation of an optionsprocessor

### CODE REVIEW

**Comment by JensWehner** [comment]
@votca-bot format

**Comment by JensWehner** [comment]
@votca-bot changelog: added optionshandler

File: CMakeModules/FindMKL.cmake [hunk 1]
```diff
@@ -94,7 +94,6 @@ find_path(MKL_INCLUDE_DIR
   PATHS
     /opt/intel
     /opt/intel/mkl
-    /opt/intel/oneapi/mkl/latest
```
Comment (junghans) [inline CMakeModules/FindMKL.cmake:97]: Why?

Comment (JensWehner) [reply CMakeModules/FindMKL.cmake]: merge error, have reversed that.

File: include/votca/tools/tokenizer.h [hunk 2]
```diff
@@ -53,8 +53,11 @@ inline T convert_impl(const std::string &s, type<T>) {
 inline bool convert_impl(const std::string &s, type<bool>) {
   if (s == "true" || s == "TRUE" || s == "1") {
     return true;
-  } else {
+
+  } else if (s == "false" || s == "FALSE" || s == "0") {
```
Comment (rubengerritsen) [inline include/votca/tools/tokenizer.h:57]: Maybe add a `tolower(s) == "false"` then a user can also type `False`

File: src/libtools/calculator.cc [hunk 3]
```diff
@@ -18,151 +18,7 @@
 // Local VOTCA includes
```
Comment (rubengerritsen) [inline src/libtools/calculator.cc:18]: Do we still need this file?

**Review by rubengerritsen — COMMENTED** [review]
Looks good. 

I am wondering your `Link Resolver` now only acts on the defaults, if I am not mistaken. Could it be easily extended to also work on the user provided input file. 

I know we have had the discussion about the nested options files before and that you are not a fan. But for the qmmm calculations I am trying now, it makes much sense. If you create one dft and one gwbse option file that can be used for the vaccuum calculation, parameterization and the qmmm embedding that would be great, because as a user you do not need to worry about keeping basissets, functionals and convergence parameters in sync across many files. Also it seems that with the link resolver it could be just an optional feature instead of mandatory nesting.

File: src/libtools/calculator.cc (hunk 3 above)
Comment (JensWehner) [reply src/libtools/calculator.cc:18]: Not really, but on the other hand it is always good to not have to create it again and it does no harm, so I would leave it.

File: include/votca/tools/tokenizer.h (hunk 2 above)
Comment (JensWehner) [reply include/votca/tools/tokenizer.h:57]: good point thanks

**Comment by JensWehner** [comment]
Concerning having nested input files, yes technically it is no big problem, but it is not so easy, because we atm demand that options are inside the value of the xml for input files.

**Review by rubengerritsen — APPROVED** [review]

**Comment by rubengerritsen** [comment]
Apparently my review is useless on this repo, but we can take up the nested options at a later time if we really want it

**Comment by JensWehner** [comment]
@votca-bot format

**Review by rubengerritsen — APPROVED** [review]

**Comment by JensWehner** [comment]
@votca-bot format

**Pull request merged**